### PR TITLE
Add report controller, service and delegate AB#10776

### PR DESCRIPTION
**Pull request by tiagograf-AOT**
# Fixes or Implements [AB#10776](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/10776)

-   [x] Enhancement
-   [ ] Bug
-   [ ] Documentation

## Description

Adds backend code for reports/CDogs

If you are reviewing this PR, please adhere to the guidelines as [documented](https://github.com/bcgov/healthgateway/wiki/prguidance).

## Testing

-   [x] Unit Tests Updated
-   [ ] Functional Tests Updated
-   [ ] Not Required

### Steps to Reproduce

If this is a bug, please provide details on how to reproduce the code.

### UI Changes

YES | NO

### Browsers Tested

-   [ ] Chrome
-   [ ] Safari
-   [ ] Edge
-   [ ] Firefox

Provide an explanation for any test(s) not completed.

## Notes/Additional Comments

Any additional notes or comments that may be relevant. Include any specialized deployment steps here.

### CODE REVIEW

File: Apps/WebClient/src/Server/Delegates/CDogsDelegate.cs [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+                };
+
+                this.serviceEndpoint = new Uri(StringManipulator.Replace(this.cdogsConfig.BaseEndpoint, replacementData) !);
+            }
+            else
+            {
+                this.serviceEndpoint = new Uri(this.cdogsConfig.BaseEndpoint);
+            }
+
+            logger.LogInformation($"ODR Proxy URL resolved as {this.cdogsConfig.ToString()}");
```
Comment (sslaws) [inline Apps/WebClient/src/Server/Delegates/CDogsDelegate.cs:83]: CDogs URL...

File: Apps/WebClient/src/appsettings.Development.json [hunk 2]
```diff
@@ -97,5 +97,9 @@
         "agentApiUrl": "http://localhost:8024/",
         "agentApiKey": "agent-api-key-dev",
         "webhookApiKey": "WebHookApiKey"
+    },
+    "CDOGS": {
+        "DynamicServiceLookup": "true",
```
Comment (sslaws) [inline Apps/WebClient/src/appsettings.Development.json:102]: I think this should be false

**Review by sslaws — CHANGES_REQUESTED** [review]
A couple of minor things found

File: Apps/WebClient/src/appsettings.Development.json (hunk 2 above)
Comment (tiagograf-AOT) [reply Apps/WebClient/src/appsettings.Development.json]: it works as is but i changed it anyway

**Review by MikeLyttle — APPROVED** [review]

**Review by sslaws — APPROVED** [review]

**Pull request merged**